### PR TITLE
new: new field 'input'

### PR DESCRIPTION
**Pull request by lionel-bijaoui**
#50 this is my proposal for this fields.
The test are not amazing for now (they accept any attribute regardless of the input type).

### CODE REVIEW

**Comment by icebob** [comment]
Thank you for your PR. I need more time to check it, ok?

**Comment by icebob** [comment]
I think, we can't handle all types with one field. Because date* types there are extra properties and methods to format date to the input and to the model like [here](https://github.com/icebob/vue-form-generator/blob/master/src/fields/fieldDateTime.vue#L28)

Do you have any idea how to solve it?

**Comment by lionel-bijaoui** [comment]
A simple check to see the this.schema.format could be enough:
```js
if (typeof this.schema.format !== "undefined" && value != null) {
	return moment(value, this.schema.format).format(this.getDateFormat());
}
return value;
```
If the user don't want to format the date or if there is no reason to use format (for non date like input), it simply return the raw value.

**Comment by icebob** [comment]
Not enough, because current date fields (datePicker, pickaday) are stored the data in Unix format without `format`. If you set `format` property it will store as a formatted date `string`. But HTML5 date* field raw format is string. So we always should format this raw values.

I created a fiddle to check the raw values of HTML5 date fields: https://jsfiddle.net/ndz05ku4/2/

**Comment by lionel-bijaoui** [comment]
What about using `inputType` ?
```js
if (value != null) {
    if (this.schema.inputType === date ||
        this.schema.inputType === datetime ||
        this.schema.inputType === datetimelocal ||
        this.schema.inputType === time ||
        this.schema.inputType === week ||
        this.schema.inputType === month) {
            if (typeof this.schema.format === "undefined"){
                // Unix formating
            }else{
                return moment(value, this.schema.format).format(this.getDateFormat());
            }
    }
}
return value;
```

**Comment by icebob** [comment]
Ok, but skip `month` and `week`. They are not full date, can't be format with moment. Stays in string.

**Comment by lionel-bijaoui** [comment]
I'm skipping `time` too

**Comment by lionel-bijaoui** [comment]
Progress: `date`, `datetime` and `datetime-local` can use the same model as Pikaday, dateTime (with date format) without problem. Modifying one or the other end in the same model format (UNIX) and update value.
`time` work well with dateTime (with time format), although Chrome don't allow to input seconds (but display them if set). Edge is fine with seconds.
`month` and `week` are treated as string.

**Comment by icebob** [comment]
Great thank you!

**Comment by lionel-bijaoui** [comment]
@icebob Can I add a new schema to the commit ?
I have reorganize the fields so that it is easier to use
[screenshot](https://cloud.githubusercontent.com/assets/3472596/18639210/3f4ba972-7e93-11e6-81f4-fb3c6a901dcb.png)

**Comment by icebob** [comment]
Sure :)

**Comment by lionel-bijaoui** [comment]
@icebob I think it is ready to merge, no ?

File: src/fields/fieldInput.vue [hunk 1]
```diff
@@ -37,7 +37,35 @@
...
+					case "datetime":
+						return moment(value).format();
+					case "datetime-local":
+						return moment(value).format("YYYY-MM-DDTHH:mm:ss");
+					default:
+						return value;
+				}
+			},
+			formatValueToModel(value) {
+				console.log(this.schema.inputType, typeof value);
```
Comment (icebob) [inline src/fields/fieldInput.vue]: Please remove this log message and after it I will merge.

Comment (lionel-bijaoui) [inline src/fields/fieldInput.vue]: done

Comment (icebob) [inline src/fields/fieldInput.vue]: Thanks

Comment (lionel-bijaoui) [inline src/fields/fieldInput.vue]: You still need to merge, no ?

Comment (icebob) [inline src/fields/fieldInput.vue]: Just waited the green lights :)

Comment (lionel-bijaoui) [inline src/fields/fieldInput.vue]: sorry I did not get it :+1:

**Pull request merged**